File: android/src/main/java/com/iconshot/detonator/request/TextAreaSetValueRequest.java
package com.iconshot.detonator.request;

import android.widget.EditText;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.tree.Edge;

public class TextAreaSetValueRequest extends Request<String> {
    public TextAreaSetValueRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    @Override
    public void run() {
        String value = decode(String.class);

        Edge edge = getComponentEdge();

        Edge elementEdge = edge.children.get(0);

        EditText view = (EditText) elementEdge.element.view;

        view.setText(value);
    }
}

File: ios/Sources/Detonator/Request/TextAreaSetValueRequest.swift
import UIKit

class TextAreaSetValueRequest: Request {
    override public func run() {
        let value: String! = decodeData()
        
        let edge = getComponentEdge()!
        
        let elementEdge = edge.children[0]
        
        let element = elementEdge.element! as! TextAreaElement
        
        element.updateValue(value: value)
        
        end()
    }
}
